viv_utils/emulator_drivers.py

```diff
@@ -236,7 +236,9 @@ def _handle_hook(self):
             # note that we prefer locally configured hooks, first.
             hook = emu.hooks.get(callname)
             try:
-                hook(self, api, argv)
+                # the emulator hook API differs
+                # TODO keep hook API here consistent with vivisect API?
+                hook(self, callconv, api, argv)
```

Comment:
`callconv` can be fetched from within `api`, so i think this is an extra argument

Reply:
as for the compatibility with viv thing... is there any cross pollination between the two hooks? i didn't think so, but i also didn't look closely. if not, my initial thinking was to keep things separate and not try to mix them too much. and, therefore, its an opportunity to clean up the viv-style hooks a bit.

Reply:
ah, this is the viv hooking subsystem. yeah, lets continue to use their format, sorry!